server/base/src/main/java/org/apache/accumulo/server/master/balancer/RegexGroupBalancer.java

```diff
@@ -68,7 +69,8 @@ protected long getWaitTime() {
   @Override
   protected Function<KeyExtent,String> getPartitioner() {
 
-    Map<String,String> customProps = configuration.getTableConfiguration(tableId).getAllPropertiesWithPrefix(Property.TABLE_ARBITRARY_PROP_PREFIX);
+    Map<String,String> customProps = tabletBalancerEnvironment.getConfiguration().getTableConfiguration(tableId)
```

Comment:
Should probably abstract some of this into the environment, because the pluggable implementation probably shouldn't have access to the internal TableConfiguration objects directly. I'm working on changing the AccumuloConfiguration class hierarchy and it is likely to impact how it is used here. Returning the per-table configuration as a simple `Map<String, String>` from the environment would make it more resilient to changes in implementation detail.